pkg/network/http/ebpf_ssl.go

```diff
@@ -327,3 +299,13 @@ func (o *sslProgram) httpsSupported() bool {
 
 	return kversion >= kernel.VersionCode(5, 5, 0)
 }
+
+func sysOpenAt2Supported() bool {
+	kversion, err := kernel.HostVersion()
+	if err != nil {
+		log.Error("could not determine the current kernel version. fallback to do_sys_open")
+		return false
+	}
+
+	return kversion >= kernel.VersionCode(5, 6, 0)
```

Comment:
Alternatively, we have ways to check if the symbol is defined in the kernel. Would that be better?

Reply:
I change the logic to check the symbol in /proc/kallsyms and if /proc is not accessible, fallback to check the kernel version.
I took a look at lxr (5.6, 5.7, ... 5.18) and sys_openat2() exist from 5.6.0 and is not under ifdef, and sys_open() call sys_openat2()